### PR TITLE
File picker undo should work

### LINKED ISSUE #69536: Simplified file picker undo doesn't work as expected
image

ctrl+backspace

image

ctrl+backspace

image

ctrl+z

image

ctrl+z

image

I see two ways we could fix:

1. Don't autocomplete when ctrl+z is pressed
2. Remove the selection before applying the undo (probably better)

Ideally redo would also function correctly as well, right now redo will leave the duplicated "REDMOND" part

### PR DESCRIPTION
**Pull request by alexr00**
It turns out that updates made to the the input value programatically are not added to the undo buffer. Any time the user makes a change to the input box value, we need to make sure they don't have to delete the programatically added auto complete value because otherwise the undo buffer will contain their delete but not the programatic add. This results in unusable undo. Fixes #69536

### CODE REVIEW

**Comment by alexr00** [comment]
@chrmarti I am definitely open to a better fix than this. I've just spent several days trying to come up with a better option, but I haven't found one.

File: src/vs/workbench/browser/parts/quickinput/quickInput.ts [hunk 1]
```diff
@@ -530,6 +537,20 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 								this.ui.list.domFocus();
 							}
 							break;
+						case KeyCode.Enter:
+						case KeyCode.Ctrl:
+						case KeyCode.Shift:
+						case KeyCode.LeftArrow:
```
Comment (chrmarti) [inline src/vs/workbench/browser/parts/quickinput/quickInput.ts]: Is there a better way to know if the key will update the undo history? Should Alt be part of this list?

Does this work as expected when the user presses right arrow and then a letter?

**Comment by alexr00** [comment]
@chrmarti I have re-implemented the auto complete using execCommand to insert text.

File: src/vs/workbench/browser/parts/quickinput/quickInput.ts [hunk 2]
```diff
@@ -463,6 +463,10 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 		this.update();
 	}
 
+	public inputHasFocus(): boolean {
+		return this.ui.inputBox.hasFocus();
```
Comment (chrmarti) [inline src/vs/workbench/browser/parts/quickinput/quickInput.ts]: This needs to check `this.visible` first.

File: extensions/css/cgmanifest.json [hunk 3]
```diff
@@ -4,13 +4,13 @@
 			"component": {
 				"type": "git",
 				"git": {
-					"name": "atom/language-css",
-					"repositoryUrl": "https://github.com/atom/language-css",
-					"commitHash": "7e06c88b89218fe6e8eba77fb674152f1cea0b10"
+					"name": "octref/language-css",
```
[the remark this answers is not in the document]
Comment (alexr00) [reply extensions/css/cgmanifest.json]: It was intentional, I just pushed to the wrong branch.

[the remark this answers is not in the document]
Comment (alexr00) [reply extensions/css/cgmanifest.json]: I reverted it.

**Pull request merged**